packages/account/src/Sections/Security/Passkeys/components/__tests__/passkey-modal.spec.tsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import PasskeyModal from '../passkey-modal';
+
+describe('PasskeyModal', () => {
+    let modal_root_el: HTMLElement;
+    beforeAll(() => {
+        modal_root_el = document.createElement('div');
+        modal_root_el.setAttribute('id', 'modal_root');
+        document.body.appendChild(modal_root_el);
+    });
+
+    afterAll(() => {
+        document.body.removeChild(modal_root_el);
+    });
+
+    it('renders the modal correctly and responds to user interaction', () => {
+        const handleClick = jest.fn();
+        render(
+            <PasskeyModal
+                has_close_icon
+                is_modal_open
+                header={<span>Test Header</span>}
+                description={<span>Test Description</span>}
+                button_text={<span>Test Button</span>}
+                onButtonClick={handleClick}
+            />
+        );
+
+        const close_icon = screen.getByTestId('dt_modal_close_icon');
+        expect(screen.getByText('Test Header')).toBeInTheDocument();
```

Comment:
I always suggest putting reused strings like 'Test Header', 'Test Description' and others into variables, so that if someone decided to change them, it can be done in one place, not in multiple places - to enhance maintainability of the test file. :)